cypress-io/cypress
# feat: utilize electron fuses to ensure that the v8 snapshot is not included in the renderer process

- Closes #22985

### User facing changelog


n/a

### Additional details


Due to an existing iframe memory leak and the fact that our V8 snapshot ended up getting instantiated inside of every V8 context (including every memory leak), we would crash out of memory when Cypress tests were run in the electron browser. We [worked with electron](https://github.com/electron/electron/issues/35170) to ensure that V8 snapshots would only be included in the main process. This PR makes the changes needed to utilize the functionality added in the linked electron PR. Specifically we need to use an electron fuse to ensure that we are reading our custom v8 snapshot on the main process (while keeping the default snapshot on other processes)

### Steps to test


Mostly this is tested via the v8 snapshot smoke test during binary creation. To test this manually, you can launch cypress in open mode and ensure that `window.snapshotResult` is undefined

### How has the user experience changed?


n/a

### PR Tasks


- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
